### PR TITLE
[MetricsAdvisor] Stop ignoring listing live tests

### LINKED ISSUE #18004: [MetricsAdvisor] Investigate flaky live tests
- GetDataFeeds: https://dev.azure.com/azure-sdk/internal/_build/results?buildId=688945&view=logs&j=9e4810bd-6c8e-5aff-29e0-d5fb4fb1935f&t=aebe1dc3-650d-52a7-62d2-615892e27a94&l=284
- GetHooksWithMinimumSetup: https://dev.azure.com/azure-sdk/internal/_build/results?buildId=688945&view=logs&j=45781f16-5afd-5a21-8819-a80ba864d80d&t=c2d7e93d-4dab-5a67-f90b-60eafdbcd41c&l=491
- GetAlertConfigurations: https://dev.azure.com/azure-sdk/internal/_build/results?buildId=698137&view=logs&j=45781f16-5afd-5a21-8819-a80ba864d80d&t=c2d7e93d-4dab-5a67-f90b-60eafdbcd41c&l=134
- GetDetectionConfigurations: https://dev.azure.com/azure-sdk/internal/_build/results?buildId=696150&view=logs&j=45781f16-5afd-5a21-8819-a80ba864d80d&t=c2d7e93d-4dab-5a67-f90b-60eafdbcd41c&l=196

### PR DESCRIPTION
**Pull request by kinelski**
Enabling some tests (fixes https://github.com/azure/azure-sdk-for-net/issues/18004). The issues in these "listing" APIs were caused by race conditions in the service. According to the service team, these bugs have been fixed.

It's been a long time since we disabled these tests, so needed to re-record two of them.

### CODE REVIEW

File: sdk/metricsadvisor/Azure.AI.MetricsAdvisor/tests/MetricsAdvisorAdministrationClient/NotificationHookLiveTests.cs [hunk 1]
```diff
@@ -345,13 +345,12 @@ public async Task UpdateWebNotificationHookWithEveryMember()
...
         [TestCase(true)]
         [TestCase(false)]
-        [Ignore("https://github.com/Azure/azure-sdk-for-net/issues/18004")]
         public async Task GetHooksWithMinimumSetup(bool useTokenCredential)
         {
             MetricsAdvisorAdministrationClient adminClient = GetMetricsAdvisorAdministrationClient(useTokenCredential);
 
             string hookName = Recording.GenerateAlphaNumericId("hook");
-            var hookToCreate = new WebNotificationHook(hookName, new Uri("http://contoso.com/"));
+            var hookToCreate = new EmailNotificationHook(hookName) { EmailsToAlert = { "fake@email.com" } };
```
Comment (kinelski) [inline sdk/metricsadvisor/Azure.AI.MetricsAdvisor/tests/MetricsAdvisorAdministrationClient/NotificationHookLiveTests.cs:353]: It doesn't really matter here if we have a `WebNotificationHook` or an `EmailNotificationHook`, as long as we have a hook. We can't really test web hooks right now (https://github.com/Azure/azure-sdk-for-net/issues/21504), so updated this one.

**Review by maririos — APPROVED** [review]

**Pull request merged**